### DISCUSSION TITLE
Problem with finding checkbox element

### DISCUSSION THREAD

**Discussion by stefanzvkvc** [keathley/wallaby#696] [Q&A]
Hi, 

I'm trying to write some tests using `wallaby` and currently I'm stuck with some problem and I need your help to point me in right direction.
Problem is when I want to use a function to click on checkbox
`
session = click(session, Query.checkbox("autoAccept"))
`
 I'm constantly getting error:
`Expected to find 1, visible checkbox 'autoAccept' but 0, visible checkboxes were found.`

I've inspected element with chrome dev tool and checkbox is actually there
![Screenshot 2022-07-04 at 14 07 02](https://user-images.githubusercontent.com/29733588/177151547-741f0523-3be3-4168-ab95-1552e55a4c46.png)

Do you maybe know what I'm doing wrong here?

Thanks in advance!

Stefan Zivkovic

**Comment by mhanberg** [comment]
Can you share your full test?

**Reply by stefanzvkvc** [reply]
Hi Mitchell, sorry for my late response.

What was my idea. I wanted to use `wallaby` to create script for load test by simulating realistic user interactions. Application that I wanted to test is web-based meetings solution.

So first I've started host session:
`
{:ok, host_session} = Wallaby.start_session(capabilities: capabilities)
`

Then I'm logging as a host and I want to turn on the option for auto-accept guests:
```
 host_session =
    host_session
    |> visit(environment)
    |> fill_in(Query.text_field("username"), with: username)
    |> fill_in(Query.text_field("password"), with: password)
    |> click(Query.button("Login"))
    # |> click(Query.checkbox("autoAccept")) <---- here is point of break
```

When host logged in, next step is to create guests sessions and join the meeting room:
```
guest_sessions =
  for number <- 1..guest_number do
    # Starts browser session for guest
    {:ok, guest_session} = Wallaby.start_session(capabilities: capabilities)
    # Joins waiting room as a guest
    guest_session =
      guest_session
      |> visit(environment)
      |> click(Query.button("Join as a guest"))
      |> fill_in(Query.text_field("roomCode"), with: room_code)
      |> click(Query.button("Join"))
      |> fill_in(Query.text_field("userName"), with: "guest#{inspect(number)}")
      |> click(Query.button("Next step"))
      |> click(Query.button("Join"))

      # Separates the request to start a new browser session
      Process.sleep(thinktime)
      guest_session
  end
```

Because I couldn’t use `autoAccept` option by clicking on specific button as a host, I found workaround.
Whenever guest joins room, it will be appended in list, so I will iterate through elements and for each of them I will use `click/2` with host session to accept these guests.

```
host_session =
  for number <- 1..guest_number, reduce: host_session do
    host_session ->
      click(host_session, Query.button("acceptRequestForguest#{inspect(number)}"))
  end
```

At glance, I thought host session needs some time after `Login` before I click on autoAccept button so I put `Process.sleep/1` but that didn't help me.
